src/main/java/org/jenkinsci/plugins/durabletask/ProcessLiveness.java

```diff
@@ -43,6 +54,23 @@
      * @return true if it is apparently still alive (or we cannot tell); false if it is believed to not be running
      */
     public static boolean isAlive(VirtualChannel channel, int pid, Launcher launcher) throws IOException, InterruptedException {
+        Boolean working = workingLaunchers.get(launcher);
+        if (working == null) {
+            // Check to see if our logic correctly reports that an unlikely PID is not running.
+            working = !_isAlive(channel, 9999, launcher);
```

Comment:
Yes it is, but it _unlikely_ to be a running process. If it is, when the first process is run on this slave, then `ProcessLiveness` will simply be disabled—which only matters in case the controlling `sh` process is killed for some weird reason, or the computer is rebooted. This reasoning is explained in JIRA.

I could have chosen an even less-likely integer but I wanted to choose a _potentially_ valid PID. Linux normally caps PIDs at 32k, and I do not wish to experiment with the effect of calling `getpgid` on an argument outside the bounds of what the kernel considers valid PIDs. (From what I can tell it correctly returns -1 on Linux.) I am not sure if there is a portable way to tell what `pid_t` is.